src/lock_file.cpp

```diff
@@ -89,6 +90,12 @@ LockFile::LockFile(const char* path_, PassKey) : path(path_)
 
 LockFile& LockFile::Get(const char* path)
 {
+#ifdef _WIN32
+    // The character ':' is reserved on Windows and cannot be used for constructing
+    // a path except when it follows a drive letter.
+    std::string file{std::regex_replace(path, std::regex(":memory:"), "memory_")};
```

Comment:
[Recommendation] @apwojcik AFAICS the following special symbols are allowed for the all filesystems of interest:
- `# $ % & ' ( ) * + - ^ _  { } ~`.

If we see disallowed symbol, then we can translate it to something like `{{decimal-code}}`. For ASCII colon, this gives `{{58}}`. This would reduce probability of accidental matching to some other file.

Reply:
Unfortunately, we cannot replace it based on the character. For example, for the path `C:\Some\Path\:memory:\and\the_rest.lock`, the first `:` character has to stay. The other occurrences should be replaced. However, the "driver letter" may actually be a multi-letter signature (e.g., `AZ:\`) for large storage systems.
It is even more complicated because the path may use UNC format: `\\?\C\Some\Path\:memory:\and\the_rest.lock`, etc. So, just finding the first occurrence will not work. First, we need to know the format used; things are getting complicated quickly. The easiest way is to create paths only with charters allowed on both Linux and Windows.

Reply:
> * `# $ % & ' ( ) * + - ^ _  { } ~`.
That is only when using NTFS. On FAT those are not allowed.

Reply:
Please see comment above